transaction/core/src/tx_error.rs

```diff
@@ -65,6 +66,12 @@ pub enum NewMemoError {
     OutputsAfterChange,
     /// Changing the fee after the change output is not supported
     FeeAfterChange,
+    /// Invalid recipient address
+    InvalidRecipient(PublicAddress),
```

Comment:
Generally I don't think we have treated public addresses as sensitive. I think there is some value in leaving it here since it will make debugging a bit easier if the code path that results in this error does trigger, but it's pretty small so I am also happy to remove it. @garbageslam @jcape any thoughts on this?

Reply:
So I remember, at some point a few years ago, there was a lot of discussion with Kyle about, are our error messages safe to log, particularly because, Signal is very concerned about any PII in their debug log files.

I remember that at one point, there was an idea that we should audit everything we are logging and make sure nothing bad can ever end up in an error message. I'm not sure if that happened. I think it's also possible that direct `Display` of mobilecoin error messages is not now logged by the signal client, because we weren't sure. Again, I'm not really sure what happened.

We should probably have a high level discussion about PII in error messages and try to avoid it. It would be good if we had some kind of a policy. Thanks for raising this issue @nick-mobilecoin !

Reply:
Ok, I will remove it for now.